app/src/main/java/com/keylesspalace/tusky/util/AbsoluteTimeFormatter.kt

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2022 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.util
+
+import java.text.SimpleDateFormat
+import java.time.ZoneId
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+
+public class AbsoluteTimeFormatter {
+    private val sameDaySdf = SimpleDateFormat("HH:mm", Locale.getDefault())
+    private val sameYearSdf = SimpleDateFormat("MM-dd HH:mm", Locale.getDefault())
+    private val otherYearSdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+    private val otherYearCompleteSdf = SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault())
+
+    fun format(time: Date?, shortFormat: Boolean = true): String {
+        return if (time == null) {
+            "??"
+        } else if (android.text.format.DateUtils.isToday(time.time)) {
+            sameDaySdf.format(time)
+        } else if (AbsoluteTimeFormatter.getYear(time) == (Calendar.getInstance().get(Calendar.YEAR) - 1900)) {
+            sameYearSdf.format(time)
+        } else if (shortFormat) {
+            otherYearSdf.format(time)
+        } else {
+            otherYearCompleteSdf.format(time)
+        }
+    }
+
+    companion object {
+        @JvmStatic
+        private fun getYear(time: Date): Int {
```

Comment:
Could make that an extension function `private fun Date.year(): Int`.

Also, @JvmStatic is not needed when only called from Kotlin.